### PR TITLE
docs: update README (take over #131)

### PR DESCRIPTION
**Pull request by Okabe-Junya**
take over #131 

## Proposed Changes

I have structured the README. Only the essential information is retained in the README, and I've made changes to store each document in the `docs` directory. This is based on the structure of the [CodeQL repository](https://github.com/github/codeql).

(If there are any rules or standards we must adhere to within the GitHub organization, I would like to make further adjustments to comply with them.)

## Readiness Checklist

In addition, I made the following changes:

- Removal of the Table of Contents: GitHub's markdown preview has an automatic table of contents generation feature. It's unlikely we need to maintain this ourselves.
- Removal of permalinks to each document: Most users should notice that the documents are stored in the `docs` directory. Managing permalinks can be cumbersome, so I believe it's best to remove them.

### Author/Contributor

**Note**: In this PR, since only documentation updates are being made, I don't anticipate any linting or testing issues to arise.

- [x] If documentation is needed for this change, has that been included in this pull request
- [x] run `make lint` and fix any issues that you have introduced
- [x] run `make test` and ensure you have test coverage for the lines you are introducing

### Reviewer
- [x] Label as either `bug`, `documentation`, `enhancement`, `infrastructure`, or `breaking`

### CODE REVIEW

**Comment by Okabe-Junya** [comment]
It seems there are still some points that need correction. After making the changes, I will undraft the PR.

File: README.md [hunk 1]
```diff
@@ -32,8 +12,8 @@ search for can be filtered by using a search query.
 |Time to Answer (Discussions Only) | The time from creation to an answer.|
 |Time in Label | The duration from label application to removal, requires LABELS_TO_MEASURE env variable.|
 
-*For pull requests, these metrics exclude the time the PR was in draft mode.
-*For Issue and pull requests, issue/pull request author's own comments and comments by bots are excluded.
+* For pull requests, these metrics exclude the time the PR was in draft mode.
```
Comment (Okabe-Junya) [inline README.md:16]: Is this a typo in the item enumeration?

Comment (zkoppert) [reply README.md:16]: these are intended to be footnotes

File: README.md [hunk 2]
```diff
@@ -1,51 +1,35 @@
...
-- As a maintainer, I want to see metrics for issues and pull requests on the repository I maintain in order to ensure I am giving them the proper amount of attention.
-- As a first responder on a repository, I want to ensure that users are getting contact from me in a reasonable amount of time.
-- As an OSPO, I want to see how many open source repository requests are open/closed, and metrics for how long it takes to get through the open source process.
-- As a product development team, I want to see metrics around how long pull request reviews are taking, so that we can reflect on that data during retrospectives.
+| Metric | Description |
+|--------|-------------|
+|Maintainer Metrics| Monitor issue and pull request metrics for effective attention.|
+|Timely Response| Ensure prompt user contact as a first responder.|
+|OSPO Insights| Track open source request status and processing times.|
+|Review Efficiency| Analyze PR review durations for retrospectives.|
```
Comment (zkoppert) [inline README.md:32]: I prefer to keep the original here. I feel it better lines up with software industry's use case language.

**Comment by zkoppert** [comment]
it looks like a few of the docs like the json output, measuring time and others in the docs directory aren't all referenced in the  README. I think each doc should at least get a mention in the readme.

**Comment by Okabe-Junya** [comment]
Thank you for the wonderful review!!

Regarding "I think each doc should at least get a mention in the readme.", are you suggesting adding a section with links to each item as in the following example?

```markdown
## links

- []
```

**Comment by zkoppert** [comment]
Yes that would be great.

File: README.md (hunk 1 above)
Comment (Okabe-Junya) [reply README.md]: I've reverted the changes to "Example use cases."  ( https://github.com/github/issue-metrics/pull/139/commits/659c287446fda32d1d830c23deaf678353c93533 )

File: README.md (hunk 2 above)
Comment (Okabe-Junya) [reply README.md]: I also thought this aligned with the use case. I've reverted the changes.

https://github.com/github/issue-metrics/pull/139/commits/1b6d343111bc79863010ae787e429f9f23d62163

**Comment by Okabe-Junya** [comment]
Thank you for the wonderful review!! I made three changes. 

1. I've reverted the changes to the footnote.
2. I've reverted the changes to the example use case.
3. I've added a section with links to each document in the README.

**Review by zkoppert — APPROVED** [review]
Awesome! Thank you so much @Okabe-Junya for leading this to completion!! Really appreciate all your contributions to this project! 🎉 🚀

**Comment by spier** [comment]
@zkoppert looks like merging this has accidentally done some of the changes that we did in https://github.com/github/issue-metrics/pull/137 prior to this.

**Comment by zkoppert** [comment]
@spier Good Catch! I'll get that corrected. https://github.com/github/issue-metrics/pull/140

**Pull request merged**